core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1547,11 +1577,14 @@ class ReplicaManager(val config: KafkaConfig,
               followerStartOffset = readResult.followerLogStartOffset,
               followerFetchTimeMs = readResult.fetchTimeMs,
               leaderEndOffset = readResult.leaderLogEndOffset)) {
+              if (partition.isAddingReplica(followerId)) {
+                followerReassignmentLagStats.updateLag(partition.topicPartition, Math.max(0L, readResult.highWatermark - readResult.info.fetchOffsetMetadata.messageOffset))
```

Comment:
If we update this only in `updateFollowerFetchState`, then it's not much better than the follower metric. We'd also want to update it when we accept an append.